tutorials/scripting/c_sharp/c_sharp_basics.rst

```diff
@@ -388,3 +388,217 @@ The following tools may be used for performance and memory profiling of your man
 - Visual Studio.
 
 Profiling managed and unmanaged code at once is possible with both JetBrains tools and Visual Studio, but limited to Windows.
+
+Using ``async``/``await``
+-----------------
+
+You might face a scenario where you must ``await`` a method call.
+You will notice that when you use ``await``, you are required to mark the method you use it in as ``async``, and change the return type to ``Task`` or ``Task<T>``. 
+Consequently, you must call your now ``async`` method using ``await``, which propogates the problem all the way up the call chain.
+This is why many people compare ``async``/``await`` to a "zombie virus", because it tends to spread once introduced.
+
+In Godot, the conclusion to this spread is the entry point methods of a node, such as ``_Ready()`` or ``_Process()``.
+You will notice that the return types of these methods are ``void`` rather than ``Task``.
+It is considered conventional wisdom in C# to avoid ``async void`` at all times, with the exception of event handlers.
+The problem is that it is impossible to change the signatures of these methods since they are defined by the classes they inherit.
+
+There are a couple options to address this problem, but each option comes with its own caveats and considerations. To compare these options, we will work with the following script:
+
+.. code-block:: csharp
+
+    using Godot;
+    using System;
+    using System.Threading.Tasks;
+
+    public partial class AsyncTestNode : Node
+    {
+        int _taskCount = 0;
```

Comment:
We always explicitly type access modifiers. Ditto for all implicitly private fields.